internal/services/apicenter/api_center_service_resource.go

```diff
@@ -0,0 +1,217 @@
+				Name:     &model.Name,
+				Location: model.Location,
+				Tags:     &model.Tags,
+				Identity: expandedIdentity,
+			}
+
+			if _, err = client.CreateOrUpdate(ctx, id, apiCenterService); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r ApiCenterServiceResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.ServicesClient
+
+			id, err := services.ParseServiceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state ApiCenterServiceResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("reading %s: %v", id, err)
+			}
+
+			if metadata.ResourceData.HasChange("identity") {
+				// TODO: Switch this to 'identity.ExpandSystemOrSingleUserAssignedMap(metadata.ResourceData.Get("identity").([]interface{}))'
+				// once SDK Helpers PR #164 has been merged and integrated into the provider...
+				identityValue, err := identity.ExpandLegacySystemAndUserAssignedMap(metadata.ResourceData.Get("identity").([]interface{}))
+				if err != nil {
+					return fmt.Errorf("expanding `identity`: %+v", err)
+				}
+
+				existing.Model.Identity = identityValue
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				existing.Model.Tags = &state.Tags
+			}
+
+			if _, err = client.CreateOrUpdate(ctx, *id, *existing.Model); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r ApiCenterServiceResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiCenter.ServicesClient
+
+			id, err := services.ParseServiceID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := ApiCenterServiceResourceModel{
+				Name:          id.ServiceName,
+				Location:      location.Normalize(resp.Model.Location),
+				ResourceGroup: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				if model.Tags != nil {
+					state.Tags = *model.Tags
+				}
+
+				if model.Identity != nil {
+					identityValue, err := identity.FlattenLegacySystemAndUserAssignedMap(model.Identity)
+					if err != nil {
+						return fmt.Errorf("flattening `identity`: %+v", err)
+					}
+
+					if err := metadata.ResourceData.Set("identity", identityValue); err != nil {
+						return fmt.Errorf("setting `identity`: %+v", err)
+					}
+				}
```

Comment:
This function accepts a pointer and handles nil values so we don't need to nil check it
```suggestion
				identityValue, err := identity.FlattenLegacySystemAndUserAssignedMap(model.Identity)
				if err != nil {
					return fmt.Errorf("flattening `identity`: %+v", err)
				}
				
				state.Identity = pointer.From(identityValue)
```